src/libc/string.rs

```diff
@@ -111,6 +111,71 @@ fn strcmp(env: &mut Environment, a: ConstPtr<u8>, b: ConstPtr<u8>) -> i32 {
     }
 }
 
+fn strncmp(env: &mut Environment, a: ConstPtr<u8>, b: ConstPtr<u8>, n: GuestUSize) -> i32 {
+    let mut offset = 0;
+    loop {
```

Comment:
I think you need to add

```rust
if n == 0 {
    return 0;
}
```

before this loop, to ensure no characters get compared in that case.